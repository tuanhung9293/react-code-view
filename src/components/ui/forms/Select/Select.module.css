@value colors: 'styles/colors.css';
@value secondary, separator, error, field from colors;

.wrapper{
  position: relative;
  margin: 10px 0;
}

.select{
  box-sizing: border-box;
  display: block;
  padding: 10px 30px 10px 15px;
  width: 100%;
  border: 2px solid transparent;
  box-shadow: 0 2px rgba(0,0,0,0.1);
  border-radius: 4px;
  background-color: field;
  appearance: none;
  background-image: url('../../../../assets/down-arrow.svg');
  background-size: 14px 7px;
  background-repeat: no-repeat;
  background-position: right 15px center;
}

.select::-ms-expand {
    display: none;
}

.error{
  border: 2px solid error;
}

.select:focus, .error:focus{
  outline: none;
  border: 2px solid secondary;
}

.slim {
  padding: 5px 35px 5px 10px;
  font-size: 0.875rem !important;
  background-position: right 10px center;
}