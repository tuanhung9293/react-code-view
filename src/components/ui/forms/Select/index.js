import React from 'react'
import classNames from 'classnames'
import styles from './Select.module.css'

function Select({hasError, value, slim, ...props}) {
  const className = classNames({
    [styles.select]: true,
    [styles.error]: hasError,
    [styles.slim]: slim
  })
  return (
    <div className={styles.wrapper}>
      <select value={value || ''} {...props} className={className} />
    </div>
  )
}

export default Select