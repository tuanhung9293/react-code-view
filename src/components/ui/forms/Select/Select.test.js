import React from 'react'
import {shallow} from 'enzyme'
jest.mock('./Select.module.css', () => ({
  wrapper: 'wrapper',
  select: 'select',
  error: 'error',
  slim: 'slim'
}))
import Select from './'

it('renders children', () => {
  expect(shallow(
    <Select>
      <option>One</option>
      <option>Two</option>
    </Select>
  )).toMatchSnapshot()
})

it('renders correctly when hasError', () => {
  expect(shallow(
    <Select hasError={true}>
      <option>One</option>
      <option>Two</option>
    </Select>
  )).toMatchSnapshot()
})