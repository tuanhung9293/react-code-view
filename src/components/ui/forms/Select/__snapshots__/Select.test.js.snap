// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children 1`] = `
<div
  className="wrapper"
>
  <select
    className="select"
    value=""
  >
    <option>
      One
    </option>
    <option>
      Two
    </option>
  </select>
</div>
`;

exports[`renders correctly when hasError 1`] = `
<div
  className="wrapper"
>
  <select
    className="select error"
    value=""
  >
    <option>
      One
    </option>
    <option>
      Two
    </option>
  </select>
</div>
`;
